reconcile/utils/saasherder/saasherder.py

```diff
@@ -1889,6 +1899,9 @@ def _validate_promotion(self, promotion: Promotion) -> bool:
         if not promotion.subscribe:
             return True
 
+        if promotion.commit_sha in self.hotfix_versions.get(promotion.url, set()):
```

Comment:
nit: can we add a small comment or dedicated func to highlight that this is a general gate switch, i.e., it must always run before any other gate is evaluated? Im afraid I might accidentally miss that in the future.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4434/commits/6f6aef7c3981af1d3c8b8d1ed335500d06ae8f42